uni/ulsp/definition.icn

```diff
@@ -0,0 +1,62 @@
+package ulsp
+
+import json
+link regexp
+link ximage
+
+class DefinitionHandler(
+   workspace,
+   params
+   )
+
+   method setVariables(new_workspace, new_params)
+      workspace := new_workspace
+      params := new_params
+   end
+
+   method run()
+      local line, character, results_table := table(),
+      wchar := &letters ++ &digits ++ '_', context, signatureHandler, possibleItems,
+      source, selectedLine := 0, i := 0, startPos := 0, endPos := 0
+
+      line := params["position"]["line"] + 1
+      character := params["position"]["character"] + 1
+
+      context := workspace.getContext(line, character)
+
+      signatureHandler := SignatureHandler()
+      signatureHandler.setVariables(workspace)
+      signatureHandler.handleNewSignature(context)
+
+      possibleItems := signatureHandler.getCurrentSigItems()
+      \possibleItems | return tojson(results_table)
+      *possibleItems > 0 | return tojson(results_table)
+      member(possibleItems[1], "source") | return tojson(results_table)
+
+      source := open(possibleItems[1]["source"])
+
+      every line := !source do {
+         if ReMatch("^[\ \t]*" || possibleItems[1]["type"] || "[\ \t]+" || possibleItems[1]["name"], line) then {
+            selectedLine := line
+            break
+         }
+         i := i + 1
+      }
+
+      close(source)
+
+      \selectedLine ? {
+         tab(many("\ \t"))
+         tab(many(wchar))
+         tab(many("\ \t"))
```

Comment:
`many()` takes a `cset`, so it it is "slightly" faster if you do `tab(many('\ \t'))` because you avoid `string` to a `cset` conversion. This is applicable to several string scanning functions. `find()` is an exception, it take a string.